digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/LanesCsvImporter.scala

```diff
@@ -0,0 +1,213 @@
+import fi.liikennevirasto.digiroad2.asset.SideCode
+import fi.liikennevirasto.digiroad2.lane._
+import fi.liikennevirasto.digiroad2.service.RoadLinkService
+import fi.liikennevirasto.digiroad2.user.User
+import fi.liikennevirasto.digiroad2.util.{LaneUtils, Track}
+import org.apache.commons.lang3.StringUtils.isBlank
+
+class LanesCsvImporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) extends CsvDataImporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) {
+  case class NotImportedData(reason: String, csvRow: String)
+  case class ImportResultLaneAsset(incompleteRows: List[IncompleteRow] = Nil,
+                                   malformedRows: List[MalformedRow] = Nil,
+                                   excludedRows: List[ExcludedRow] = Nil,
+                                   notImportedData: List[NotImportedData] = Nil,
+                                   createdData: List[ParsedProperties] = Nil)  extends ImportResult
+  type ImportResultData = ImportResultLaneAsset
+  type ParsedCsv = (MalformedParameters, List[ParsedProperties])
+
+  private val nonMandatoryFieldsMapping: Map[String, String] = Map(
+    "id" -> "id",
+    "tietyyppi" -> "road type",
+    "s_tietyyppi" -> "name road type",
+    "alkupvm" -> "date",
+    "s_katyyppi" -> "name lane type"
+  )
+
+  private val intValueFieldsMapping: Map[String, String] = Map(
+    "tie" -> "road number",
+    "ajorata" -> "track",
+    "osa" -> "road part",
+    "aet" -> "initial distance",
+    "let" -> "end distance"
+  )
+
+  private val laneNumberFieldMapping: Map[String, String] = Map("kaista" -> "lane")
+  private val laneTypeFieldMapping: Map[String, String] = Map("katyyppi" -> "lane type")
+
+  val mandatoryFieldsMapping: Map[String, String] = laneNumberFieldMapping ++ intValueFieldsMapping ++ laneTypeFieldMapping
+
+  private def findMissingParameters(csvRowWithHeaders: Map[String, String]): List[String] = {
+    mandatoryFieldsMapping.keySet.diff(csvRowWithHeaders.keys.toSet).toList
+  }
+
+  def getPropertyValue(pointAssetAttributes: ParsedProperties, propertyName: String): String = {
+    pointAssetAttributes.find(prop => prop.columnName == propertyName).map(_.value).get.asInstanceOf[String]
+  }
+
+  def getPropertyValueOption(pointAssetAttributes: ParsedProperties, propertyName: String): Option[String] = {
+    pointAssetAttributes.find(prop => prop.columnName == propertyName).map(_.value).asInstanceOf[Option[String]]
+  }
+
+  def verifyIntType(parameterName: String, parameterValue: String): ParsedRow = {
+    if (parameterValue.forall(_.isDigit)) {
+      (Nil, List(AssetProperty(columnName = intValueFieldsMapping(parameterName), value = parameterValue)))
+    } else {
+      (List(parameterName), Nil)
+    }
+  }
+
+  def verifyLaneNumber(parameterName: String, parameterValue: String): ParsedRow = {
+    val trimmedValue = parameterValue.trim
+
+    if ( Seq("11", "21", "31").contains(trimmedValue) ) {
```

Comment:
To confirm.
Is only main lanes that enter here now, because in the files you have sub lanes too, right?

Reply:
Only main lanes will be imported